src/plugins/expressions/common/execution/execution.ts

```diff
@@ -94,7 +109,10 @@ export class Execution<
   }
 
   constructor(public readonly params: ExecutionParams<ExtraContext>) {
-    const { executor, ast } = params;
+    const { executor, ast, expression } = params;
+
+    this.expression = expression || formatExpression(ast);
```

Comment:
we can put the responsibility to convert to ast to ExecutionContract ?

Reply:
seems we already do that in the executor ?

Reply:
Done. Converting to AST in exeuction.